nym-vpn-core/nym-vpnd/src/win_service/service.rs

```diff
@@ -0,0 +1,172 @@
+    }
+}
+
+fn run_service(_arguments: Vec<OsString>) -> windows_service::Result<()> {
+    info!("Setting up event handler");
+
+    let (event_tx, event_rx) = tokio::sync::mpsc::unbounded_channel();
+    let event_handler = move |control_event| -> ServiceControlHandlerResult {
+        match control_event {
+            ServiceControl::Stop => {
+                event_tx.send(()).unwrap();
+                ServiceControlHandlerResult::NoError
+            }
+            ServiceControl::Interrogate => ServiceControlHandlerResult::NoError,
+            _ => ServiceControlHandlerResult::NotImplemented,
+        }
+    };
+
+    // Register system service event handler
+    let status_handle = service_control_handler::register(SERVICE_NAME, event_handler)?;
+
+    info!("Service is starting...");
+
+    let task_manager = TaskManager::new(10).named("nym_vpnd");
+    let _service_task_client = task_manager.subscribe_named("vpn_service");
+
+    let state_changes_tx = broadcast::channel(10).0;
+
+    // The idea here for explicly starting two separate runtimes is to make sure they are properly
+    // separated. Looking ahead a little ideally it would be nice to be able for the command
+    // interface to be able to forcefully terminate the vpn if needed.
+
+    // Start the command interface that listens for commands from the outside
+    let (command_handle, _vpn_command_rx) =
+        start_command_interface(state_changes_tx.subscribe(), task_manager, None, event_rx);
+
+    // Start the VPN service that wraps the actual VPN
+    // let vpn_handle = start_vpn_service(state_changes_tx, vpn_command_rx, service_task_client);
+
+    info!("Service has started");
+
+    // Tell the system that the service is running now
+    let next_status = ServiceStatus {
+        service_type: SERVICE_TYPE,
+        current_state: ServiceState::Running,
+        controls_accepted: ServiceControlAccept::STOP,
+        exit_code: ServiceExitCode::Win32(0),
+        checkpoint: 0,
+        wait_hint: Duration::default(),
+        process_id: None,
+    };
+    status_handle.set_service_status(next_status)?;
+
+    // vpn_handle.join().unwrap();
+    command_handle.join().unwrap();
+
+    info!("Service is stopping!");
+
+    // Tell the system that service has stopped.
+    status_handle.set_service_status(ServiceStatus {
+        service_type: SERVICE_TYPE,
+        current_state: ServiceState::Stopped,
+        controls_accepted: ServiceControlAccept::empty(),
+        exit_code: ServiceExitCode::Win32(0),
+        checkpoint: 0,
+        wait_hint: Duration::default(),
+        process_id: None,
+    })?;
+
+    info!("Service has stopped!");
+
+    Ok(())
+}
+
+#[allow(unused)]
+#[derive(thiserror::Error, Debug)]
+pub enum InstallError {
+    #[error("Unable to connect to service manager")]
+    ConnectServiceManager(#[source] windows_service::Error),
+
+    #[error("Unable to create a service")]
+    CreateService(#[source] windows_service::Error),
+}
```

Comment:
Let's use a separate `error.rs` file for this, like we do in most of our codebase.